frontend/packages/console-shared/package.json

```diff
@@ -0,0 +1,10 @@
+{
+  "name": "@console/shared",
+  "version": "0.0.0-fixed",
+  "description": "Console code shared between all packages",
+  "private": true,
+  "main": "src/index.ts",
+  "scripts": {
+    "test": "yarn --cwd ../.. run test packages/console-shared"
```

Comment:
this PR is fantastic work @vojtechszocs ! ⭐️ 

My only feedback at this point is just merely a question...any chance the jest `projects` array (ProjectConfig) would be easier for you on this piece? It seems that Jest now internally supports monorepos as well.

https://jestjs.io/docs/en/configuration#projects-array-string-projectconfig

Reply:
Thank you, Patrick :smiley:

This PR retains the existing `test` script (and others) in the monorepo root `package.json` without any changes. Those scripts represent a single execution of some tool (e.g. Jest) to operate on the whole code base.

So instead of e.g. `yarn workspaces run test` (analogy to `lerna run test`), which delegates the responsibility to each package and results in multiple Jest executions, the monorepo root's `test` script simply runs Jest on all the code.

In specific packages, like `console-shared` above, we "override" the root `test` script by scoping Jest execution to that package. This is to support developer experience on the package level, where you `cd frontend/packages/console-shared` and do `yarn test` etc.

Now regarding your question, I like the idea of using Jest `projects` feature :+1: if we can support above mentioned root & per-package developer experience. Enumerating individual packages as Jest "projects" isn't good as it imposes assumptions about concrete packages, so we could go with:
```json
{
  "projects": ["<rootDir>/packages/*"]
}
```
but I think we'd still end up `cwd`-ing to monorepo root since the Jest configuration is still global:
```
yarn --cwd ../.. run test --projects packages/console-shared
```
which isn't too different from the current approach. However, I can be wrong here so let me know what you think.

The long-term idea, inspired by [fabric8-ui monorepo](https://github.com/fabric8-ui/fabric8-ui/tree/master/packages), is to extract Jest, ESLint and other tool specific configuration & scripts into a separate package (say `console-scripts`) and let each package to run those scripts while reusing the same common configuration.

So in the long-term, the root `test` script would become `yarn workspaces run test`.

Reply:
ok, thanks for the explanation @vojtechszocs - this syntax was a little odd to me, but now I understand. Agree that long-term `yarn workspaces run test` is much easier to follow and it seems this is intended ;-)
https://yarnpkg.com/lang/en/docs/cli/workspaces/#toc-yarn-workspaces-run